---
title: "Mortality Metagenomes"
output: html_notebook
---

this version with "community abundance"
```{r}
library(tidyverse)
library(phyloseq)
library(vegan)

setwd("/Users/skearney/Documents/prochlorococcus/experiments/Mortality/Data")
taxtab <- read.table("checkm_profile.txt",sep="\t",header=T)
smpname <- read.csv("BMC_template.csv")
samples <- smpname[1:89,1] 

comtab <- taxtab[,seq(7, ncol(taxtab), 4)]

smp <-sapply(samples,function(x) as.character(x))
smp[53] <- "REPC 6-HV t36"
smp[40] <- "REPC 4-LGV t36"
smpmeta <- sapply(smp,function(x) strsplit(x,split=" "))
metamat <- matrix(NA, nrow = ncol(comtab), ncol = 4)
rownames(metamat) <- smp 
colnames(metamat) <- c("Replicate","Treatment","Time","Type")
for (i in 1:length(smp)){
  if (sum(grep("G",smp[i])) > 0){
    metamat[i,4] <- "G"
  }
  else if (sum(grep("V",smp[i])) > 0){
    metamat[i,4] <- "V"
  }
  else if (sum(grep("C",smp[i])) > 0){
    metamat[i,4] <- "C"
  }
  for (j in 1:3){
    tmp <- smpmeta[[i]][j]
    metamat[i,j] <- tmp
  }
}



colnames(comtab) <- smp
rownames(comtab) <- paste0("OTU", 1:nrow(comtab))

taxa <- sapply(taxtab[,2],function(x) strsplit(as.character(x),split=";"))
taxmat = matrix(NA, nrow = nrow(comtab), ncol = 7)
rownames(taxmat) <- rownames(comtab)

colnames(taxmat) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
for (i in 1:length(tax)){
  for (j in 1:7){
    x <- as.character(strsplit(taxa[[i]][j],split="__")[[1]][2])
    taxmat[i,j] <- x
  }
}

OTU = otu_table(comtab,taxa_are_rows = T)
TAX = tax_table(taxmat)
sampledata = sample_data(data.frame(metamat))

physeq = phyloseq(OTU, TAX)
physeq1 = merge_phyloseq(physeq,sampledata)
physeq2 <- subset_samples(physeq1,samples != "REPC 3-LV t36" & samples != "REPA 8-G t48")
physeq3 <- subset_samples(physeq2,Treatment != "LG" & Treatment != "HG" & Replicate != "REPE")

P2.ord <- ordinate(physeq2, "NMDS", "jsd")
p1 = plot_ordination(physeq2, P2.ord, type="samples",shape="Treatment",color="Time")
p1 + geom_point(size=5)
p2 <- plot_richness(physeq2,measures="shannon",x="Time",color="Replicate") 
p2 + facet_wrap(~Treatment)

physeq48 <- subset_samples(physeq2, Time=="t48")
p3 <- plot_bar(physeq48,fill="Family",x="Replicate",facet_grid=~Treatment)
p3

p4 = plot_ordination(physeq2, P2.ord, type="biplot", color="Treatment", shape="Phylum")

```

this version with mapped reads (need to run first chunk first) 
using it for doing differential abundance analysis

```{r}
# note that OTUs 4, 6, 7, (19), 23 are absent from controls
# REPC 6-HV t36 seems weird... maybe swapped with REPC 4-LGV t36
library(DESeq2)
# table of the number of reads mapping to each OTU
rawtab <- taxtab[,seq(4, ncol(taxtab), 4)]

colnames(rawtab) <- colnames(comtab)
rownames(rawtab) <- rownames(comtab)

OTUr = otu_table(rawtab,taxa_are_rows = T)

physeqR = phyloseq(OTUr, TAX)
physeqR1 = merge_phyloseq(physeqR,sampledata)
physeqR2 <- subset_samples(physeqR1,samples != "REPC 3-LV t36" & samples != "REPA 8-G t48")

diagdds = phyloseq_to_deseq2(physeqR2, ~ Type)
diagdds = DESeq(diagdds, test="Wald", fitType="parametric")

resGC = results(diagdds, contrast = c("Type","G","C"), cooksCutoff = FALSE)
resGV = results(diagdds, contrast = c("Type","G","V"), cooksCutoff = FALSE)


alpha = 0.01
sigtab = res[which(res$padj < alpha), ]
sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(physeqR2)[rownames(sigtab), ], "matrix"))
head(sigtab)


```

